### ISSUE TITLE
Feature request: Filtering the text in the body

### ISSUE THREAD

**Issue by wardellbagby** [YoRyan/mailrise#63]
I'm wanting to use Mailrise to forward email to Telegram, but the emails I'm forwarding are formatted with HTML. Apprise properly converts that HTML, but that HTML also includes styling so that while it looks fine when rendered in a browser, it isn't as nice without that styling. When that HTML (and styling) is removed, the email ends up with a lot of unnecessary content.

What I'm looking for to solve that is to take an email like this:

```html
<html><h1>Here's an email!</h1><p>This is the real message!</p></html>
```

and apply a conversion such that we'd end up calling Apprise with something like:

```shell
apprise --body="This is the real message!"
```

HTML isn't regular but I think regex will likely enough for most use-cases since I imagine most people will only need to get a few lines of text from an email if they need to use it.

Something in the config like:

```yaml
configs:
  "*@*":
    mailrise:
      body_pattern: '(?<=<p>).+?(?=<\/p>)'
    urls:
      - tgram://...
```

That regex means "see text starting with `<p>` but don't include it, get all text after it, see text ending with `</p>` but don't include it". So applying that pattern to the message, we'd get my desired "This is the real message!" as the body that is sent to Apprise.

Things I've considered:
1. Putting this in Apprise instead.
    -  Feels wrong since Apprise just wants to deal with content directly, not map it to some other format.
2. Changing the email before it gets to Apprise.
    - Possible, by writing some sort of SMTP forwarder. I don't think such a thing currently exists but that's the approach I'll take if this is a poor fit for Mailrise.

Alternative approaches:
- Instead of supplying a pattern, supplying a file that can be executed with the text of the email in its args and outputs the new text as output.
  -  Seemed overly complicated but does allow the most flexibility.

It's been a while since I've written some Python, but if you think this is a useful feature, I don't mind writing it up myself. I recognize that `body_pattern` can be confusing when `body_template` already exists but I can't think of anything better.

**Comment by wardellbagby** [comment]
So I have a working branch following my initial message I'll push up in the morning for your thoughts on (it's even got tests!) but thinking more about this, maybe it's better to instead extend the existing templating logic with extra variables?

So something like:

```yaml
configs:
  "*@*":
    mailrise:
      extra_body_variables:
        - name: '(?<=<h1>).+(?=<\/h1>)
        - type: '(?<=<p>).+(?=<\/p>)'
       body_template: '$name, $type'
    urls:
      - tgram://...
```

Where then giving it an email body of:

```html
<html><h1>Wardell</h1><p>Super cool programmer</p>
```

Would then call Apprise like:

```shell
apprise --body="Wardell, Super cool programmer"
```

**Comment by YoRyan** [comment]
Ooh, I appreciate the PR, and the tests. I'm also pleased that you had a chance to work with the code that I had just finished refactoring.

But to be honest, I'm not keen on adding any additional options to the YAML configuration file. I'd like to keep that format simple, so any homelabber can hit the ground running with it—what I'd like to do instead is make it possible to supply your own Python class to determine the processing logic. (If you're the type to delve into regexes, you most likely aren't afraid of getting your hands dirty with a little Python, methinks.) In short, you'd be able to write your own SimpleRouter.

I'll have it done soon; I've just been debating the semantics. I'm thinking something like:

mailrise.conf:
```yaml
python:
  router: /path/to/my/router.py
...
```

router.py:
```python
class MyCustomRouter():
    async def email_to_apprise(self, logger, email):
        ...
        yield AppriseNotification(...)
```

I believe such custom routers will provide the most flexibility for users interested in fine-tuning their notifications.

**Comment by wardellbagby** [comment]
Hahaha, wow, great minds do think alike! 

I love your `custom_router` approach here and I'm fully okay with just waiting for that to get in. 

That being said, while I think that's a great fit for anyone building from source, it'll be a _little_ weird for anyone using Docker. Mainly because they'll likely have to check out the project anyway to get the right typings and then have their Dockerfile copy over their custom router.

Of course, it's Python and the types don't actually matter but it's going to feel a little odd to write a custom router against one version of Mailrise that might end up running in a different version. That's going to be on the users to keep in sync and I don't think it's overly-difficult, just something to remember as updates happen.

**Comment by YoRyan** [comment]
> That being said, while I think that's a great fit for anyone building from source, it'll be a _little_ weird for anyone using Docker. Mainly because they'll likely have to check out the project anyway to get the right typings and then have their Dockerfile copy over their custom router.

Not necessarily. You'll be able to passthrough the .py file just like you can do with the config file itself:

```
-v /path/to/my/config.yml:/etc/mailrise.conf
-v /path/to/my/pluggable.py:/etc/mailrise_logic.py
```

> That's going to be on the users to keep in sync and I don't think it's overly-difficult, just something to remember as updates happen.

Of course. It's important to design an API that will be unlikely to change.

**Comment by YoRyan** [comment]
Now available with b76b164271ceacf9dbaa9184d25f47eb092e4cbc. 👍
